AlfrescoApp/Model/Managers/RealmSyncManager.m

```diff
@@ -1300,7 +1300,20 @@ - (void)rebuildDataBaseWithCompletionBlock:(void (^)(BOOL completed))completionB
         }
         else
         {
-            [self handleNodeSyncActionAndStatus:node.alfrescoNode parentNode:nil];
+            //document service get node for id
+            [self.documentFolderService retrieveNodeWithIdentifier:rnode.alfrescoNode.identifier
+                                                   completionBlock:^(AlfrescoNode *node, NSError *error) {
+                if (error)
+                {
+                    [self removeTopLevelNodeFlagFomNodeWithIdentifier:[[RealmSyncCore sharedSyncCore]
+                                                                       syncIdentifierForNode:rnode.alfrescoNode]];
+                }
+                else
+                {
+                    [self handleNodeSyncActionAndStatus:rnode.alfrescoNode
```

Comment:
I think here it would be better to use the updated node received from the server. This way we would also get any updates from the server.